### PR TITLE
(SIMP-5367) Add default iptables policy rule mgmt

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Added 'iptables_default_policy' for modifying the default policy of the
  'filter' table on either IPv4 or IPv6.
- Added 'iptables::rules::default_drop' to allow users to easily toggle the
  default drop behavior of the default filter policies.
- Removed Puppet 4 support
- Added Puppet 6 support

SIMP-5367 #close

### CODE REVIEW

File: lib/puppet/type/iptables_default_policy.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    errmsg = []
+
+    if parameters[:apply_to].value.include?('ipv4')
+      v4_chains = ipv4_tables[parameters[:table].value]
+
+      if v4_chains
+        unless v4_chains.include?(parameters[:chain].value)
+          errmsg << "Invalid chain '#{parameters[:chain].value}' in table '#{parameters[:table].value}' for IPv4"
+          errmsg << "  Valid chains include '#{v4_chains.join("', '")}'"
```
Comment (sharkbruhaha) [inline lib/puppet/type/iptables_default_policy.rb]: should appear on a new line? Or at least have a period at end of previous

File: lib/puppet/type/iptables_default_policy.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        end
+      else
+        errmsg << "Invalid table '#{parameters[:table].value}' for IPv4"
+        errmsg << "  Valid tables include '#{ipv4_tables.keys.join("', '")}'"
```
Comment (sharkbruhaha) [inline lib/puppet/type/iptables_default_policy.rb]: should appear on a new line? Or at least have a period at end of previous

File: lib/puppet/type/iptables_default_policy.rb [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      end
+    end
+
+    if parameters[:apply_to].value.include?('ipv6')
+      v6_chains = ipv6_tables[parameters[:table].value]
+
+      if v6_chains
+        unless v6_chains.include?(parameters[:chain].value)
+          errmsg << "Invalid chain '#{parameters[:chain].value}' in table '#{parameters[:table].value}' for IPv6"
+          errmsg << "  Valid chains include '#{v6_chains.join("', '")}'"
```
Comment (sharkbruhaha) [inline lib/puppet/type/iptables_default_policy.rb]: should appear on a new line? Or at least have a period at end of previous

File: lib/puppet/type/iptables_default_policy.rb [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        end
+      else
+        errmsg << "Invalid table '#{parameters[:table].value}' for IPv6"
+        errmsg << "  Valid tables include '#{ipv6_tables.keys.join("', '")}'"
```
Comment (sharkbruhaha) [inline lib/puppet/type/iptables_default_policy.rb]: should appear on a new line? Or at least have a period at end of previous

File: lib/puppet/provider/iptables_default_policy/enforce.rb [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    return @needs_sync.empty?
+  end
+
+  def flush
+    if needs_sync.include?('ipv4')
+      iptables(['-t', @resource[:table], '-P', @resource[:chain], @resource[:policy]])
+    end
+
+    if needs_sync.include?('ipv6')
```
[the remark this answers is not in the document]
Comment (trevor-vaughan) [reply lib/puppet/provider/iptables_default_policy/enforce.rb]: @sharkbruhaha They call different things so there would be a lot of nasty `exec`-ing.

File: lib/puppet/provider/iptables_default_policy/enforce.rb [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+  def self.gather_resources(rules, rule_type)
+    valid_tables = ['filter']
+
+    resources = []
+
+    current_table = nil
+    rules.each do |rule|
+      if rule =~ /^\*(.*)/
+        current_table = $1.strip
```
[the remark this answers is not in the document]
Comment (trevor-vaughan) [reply lib/puppet/provider/iptables_default_policy/enforce.rb]: Don't need to do the work if it doesn't match.

File: lib/puppet/type/iptables_default_policy.rb (hunk 1 above)
Comment (trevor-vaughan) [reply lib/puppet/type/iptables_default_policy.rb]: It will be on a new line based on the join.

Comment (sharkbruhaha) [reply lib/puppet/type/iptables_default_policy.rb]: ah gotcha

**Pull request merged**